src/async.rs

```diff
@@ -356,7 +315,7 @@ where
         loop {
             let item = match self.sink_stream.poll() {
                 Ok(Async::Ready(Some(item))) => Ok(item),
-                Ok(Async::Ready(None)) => return Ok(Async::Ready(())),
+                Ok(Async::Ready(None)) => return Err(()),
```

Comment:
I'm still not really familiar with futures.
Why is `Ready(None)` turned into an error here?

Reply:
Yeah, I should have explained that. Added a comment.

Reply:
👍